python/cudf/cudf/utils/_numba_setup.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+import glob
+import os
+
+from numba import config
+
+ANY_PTX_FILE = os.path.dirname(__file__) + "/../core/udf/shim_60.ptx"
+
+
+def _setup_numba():
+    """
+    Configure numba for use with cuDF. This consists of potentially
+    putting numba into enhanced compatibility mode based on the user
+    driver and runtime versions as well as the version of the cuda
+    toolkit used to build the PTX files shipped with the user cuDF
+    package. It also sets any other config options within numba that
+    are desired for cuDF's operation.
+    """
+    _setup_numba_linker(ANY_PTX_FILE)
+    config.CUDA_LOW_OCCUPANCY_WARNINGS = 0
+
+
+def _get_best_ptx_file(archs, max_compute_capability):
+    """
+    Determine of the available PTX files which one is
+    the most recent up to and including the device cc
+    """
+    filtered_archs = [x for x in archs if x[0] <= max_compute_capability]
+    if filtered_archs:
+        return max(filtered_archs, key=lambda y: y[0])
+    else:
+        return None
+
+
+def _get_ptx_file(path, prefix):
+    if "RAPIDS_NO_INITIALIZE" in os.environ:
+        # cc=60 ptx is always built
+        cc = int(os.environ.get("STRINGS_UDF_CC", "60"))
+    else:
+        from numba import cuda
```

Comment:
Runtime imports can be expensive but importing `cuda` must be avoided as a side effect of importing the module in which this function resides. Another option could be passing the `cuda` module as an optional argument here.